src/nl/rubensten/texifyidea/util/Labels.kt

```diff
@@ -7,67 +7,84 @@ import nl.rubensten.texifyidea.index.BibtexIdIndex
-        .mapNotNull(LatexCommands::getRequiredParameters)
-        .filter { it.isNotEmpty() }
-        .map { it.first() }
+fun PsiFile.findAllLabelsInFileSet(): Sequence<String> {
+    val commands = TexifySettings.getInstance().labelCommands
+    return findLabelingCommandsSequence()
+            .map { it.name to it.requiredParameters }
+            .filter { it.second.isNotEmpty() }
+            .mapNotNull {
+                val position = commands[it.first]?.position
+                if (position != null && it.second.size >= position)
+                    it.second[position - 1] else null
+            }.filterNot { it == "" }
+}
 
 /**
  * Finds all the defined bibtex labels in the fileset of the file.
  *
  * @return A set containing all labels that are defined in the fileset of the given file.
  */
-fun PsiFile.findBibtexLabelsInFileSet(): Sequence<String> = BibtexIdIndex.getIndexedIdsInFileSet(this)
-        .asSequence()
-        .map { it.text.substringEnd(1) }
+//fun PsiFile.findBibtexLabelsInFileSet(): Sequence<String> = BibtexIdIndex.getIndexedIdsInFileSet(this)
+//        .asSequence()
+//        .map { it.text.substringEnd(1) }
+
+fun PsiFile.findBibtexLabelsInFileSet(): Sequence<String>  = findBibtexItems().asSequence()
+            .mapNotNull {
+                when (it) {
+                    is BibtexId -> it.text.substringEnd(1)
+                    is LatexCommands -> it.requiredParameter(0)
+                    else -> null
+                }
+            }
+
 
 /**
- * Finds all the labels within the collection of commands.
+ * Finds all the labeling commands within the collection of commands.
  *
  * @return A collection of all label commands.
  */
 fun Collection<PsiElement>.findLabels(): Collection<PsiElement> {
+    val commandNames = TexifySettings.getInstance().labelCommands
     return filter {
         if (it is LatexCommands) {
-            val name = it.name
-            "\\bibitem" == name || "\\label" == name
+            commandNames.containsKey(it.name)
+        }
+        else {
+            true
         }
-        else true
     }
 }
 
 /**
  * Finds all defined labels within a given file.
  *
- * @param file
- *         The file to analyse the file set of.
+ * @receiver The file to analyse the file set of.
  * @return The found label commands.
  */
-fun PsiFile.findLabels(): Collection<PsiElement> {
-    val commands = LatexCommandsIndex.getItems(this)
-    val bibtexIds = BibtexIdIndex.getIndexedIds(this)
-    val result = ArrayList<PsiElement>(commands)
-    result.addAll(bibtexIds)
-    return result.findLabels()
+fun PsiFile.findLabelingCommands(): Collection<LatexCommands> = findLabelingCommandsSequence().toList()
+
+/**
+ * make a sequence of all commands which specify an label
+ */
+fun PsiFile.findLabelingCommandsSequence(): Sequence<LatexCommands> {
+    val commandNames = TexifySettings.getInstance().labelCommands
+
+    return LatexCommandsIndex.getItemsInFileSet(this).asSequence()
```

Comment:
Why is this getItemsInFile **Set** and not `getItems` like in the removed `PsiFile.findLabels()`? It seems to be causing the crash at #895, especially when I note the comment at https://github.com/Ruben-Sten/TeXiFy-IDEA/issues/454#issuecomment-374895880

Reply:
`PsiFile.findLabels` should not be removed. Please add it back :)

Reply:
@RubenSchellekens On second thought, I do not agree. I think the new `findLabelingCommands()` _replaces_ `findLabels()`, because I think the whole point of that and this PR is to make sure the custom labeling commands are included - but @Stainzl  may correct me on this.

I believe the correct fix is replacing `getItemsInFileSet` with `getItems`, just as in the comment of that earlier crash report I mentioned.

You can reproduce the bug in this PR (just making sure there _is_ one...) by creating a file
`main.tex`:
```latex
\documentclass{article}
\begin{document}
    \input{included.tex}
\end{document}
```
`included.tex`:
```latex
\begin{figure}
    A figuring float.
    \label{fig:not-referenced}
\end{figure}
```
and you should (after maybe editing some files, you know the drill) get an exception like this one:
```
com.intellij.diagnostic.PluginException: Reported element LatexCommandsImpl(COMMANDS)[STUB]{\label} is not from the file 'LaTeX source file' the inspection 'LatexDuplicateLabel' (class nl.rubensten.texifyidea.inspections.latex.LatexDuplicateLabelInspection) was invoked for. Message: 'Duplicate label 'fig:not-referenced''.
```

I confirmed that the crashes stayed away when applying the suggested fix.

That this exception popped up for a _new_ inspection from #890 is basically a confusing coincidence I believe.

Reply:
Okay, I saw your comments in my mailbox and thougt I could answer quickly. I can confirm that my plan was to replace `findLabels()` completly and therefore I wrote a new function to check it, but don't break the whole thing. In the end I removed the old function. But as @RubenSchellekens pointed out in this [comment](https://github.com/Ruben-Sten/TeXiFy-IDEA/issues/895#issuecomment-492781202) it would be better to rename the new function back to the old name.

Reply:
I renamed the function back to the original name in commit 8e387e189139675d98b209ddcfc0198fc9186863